Public/Src/Tools/Tool.MsBuildGraphBuilder/MsBuildPredictionCollector.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using Microsoft.Build.Prediction;
+
+namespace MsBuildGraphBuilderTool
+{
+    /// <summary>
+    /// Collects predictions from the MSBuild prediction library.
+    /// </summary>
+    /// <remarks>
+    /// This validates the predicted inputs and outputs, in case the predictor is not working properly
+    /// </remarks>
+    public sealed class MsBuildPredictionCollector : IProjectPredictionCollector

```

Comment:
>MsBuildPredictionCollector [](start = 24, length = 26)

General question: All File IO Error handling is done by callers of this class?

Reply:
Yup. There's a try/catch around prediction for each build node (see `MsBuildGraphBuilder` line 363-374)

---
In reply to: [294452770](https://github.com/microsoft/BuildXL/pull/480#discussion_r294452770) [](ancestors = 294452770)